In [ ]:
%pip install pandas

In [ ]:
import pandas as pd

path_dataset = "../data/raw/dataset_original.json"

# Carga del dataset original en formato json
df = pd.read_json(path_dataset)

# Visualizacion de las primeras filas
df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


In [15]:
# Obtener las dimensiones del dataset
filas , columnas = df.shape

print(f"EL DATASET ORIGINAL TIENE")
print(f"Filas: {filas}")
print(f"Columnas: {columnas}")


EL DATASET ORIGINAL TIENE
Filas: 8160
Columnas: 8


In [17]:
# Inspeccion tecnica de tipos y nulos detectados por pandas
print("Resumen tecnico inicial (df.info()):")
df.info()
df.describe()



Resumen tecnico inicial (df.info()):
<class 'pandas.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   str    
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   str    
 5   favorite_genre            7920 non-null   str    
 6   last_login_date           7840 non-null   str    
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), str(4)
memory usage: 510.1 KB


,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.000000,8160.000000,7967.000000,8160.000000
mean,13995.433824,34.096814,1107.346153,1.800980
std,2310.810660,14.511304,5310.442604,11.334969
min,10000.000000,-5.000000,-120.000000,-1.000000
25%,11987.750000,25.000000,489.200000,0.000000
50%,13998.500000,33.000000,757.400000,1.000000
75%,15997.250000,42.000000,1045.700000,1.000000
max,17999.000000,150.000000,99999.000000,150.000000


📝 Informe de Diagnóstico Inicial y Calidad de Datos

Tras completar la radiografía técnica del dataset original en su estado crudo, se presentan las siguientes conclusiones fundamentales para el desarrollo del proyecto:
1. Estructura y Clasificación
El conjunto de datos cuenta con un volumen de 8,160 observaciones y 8 variables
. Se ha realizado una clasificación semántica que distingue identificadores (user_id), dimensiones demográficas (age, country), comportamiento del usuario (monthly_watch_time_mins, favorite_genre, last_login_date, subscription_plan) y métricas de soporte (customer_support_tickets)
.
2. Hallazgos Críticos de Calidad (Evidencia para Limpieza)
A partir de las funciones .info() y .describe(), se detectaron ruidos que comprometen la integridad del análisis si no son tratados:
Datos Faltantes: Se evidenció la ausencia de 193 registros en la variable de tiempo de visualización (monthly_watch_time_mins), lo que representa una laguna informativa que justifica el uso de técnicas de imputación o remoción en la siguiente etapa
.
Registros Lógicamente Imposibles: Se identificaron valores mínimos inválidos, como edades de -5 años y tickets de soporte con valor -1, los cuales deben ser tratados como ruido técnico
.
Sesgo por Outliers: La variable de tiempo de visualización presenta un máximo de 99,999, valor que distorsiona la media (1,107.34) respecto a la mediana (757.4), confirmando una fuerte asimetría a la derecha provocada por registros corruptos
.
3. Preguntas de Análisis / Negocio
Basado en esta inspección, el grupo se propone responder:
¿Existe una asociación entre el plan de suscripción y el volumen de visualización mensual?
¿Los usuarios con edades extremas presentan patrones de tickets de soporte distintos al usuario promedio?
¿Cómo impacta la distribución geográfica en la preferencia de géneros cinematográficos?
4. Compromiso de Trazabilidad
Se deja constancia de que en este notebook no se ha realizado ninguna modificación, eliminación ni imputación de datos
. El dataset original permanece intacto en la carpeta data/raw/ para garantizar la reproducibilidad total del proceso
. La evidencia aquí documentada servirá como el sustento obligatorio para las decisiones de limpieza que se ejecutarán en la Notebook 02

al ejecutar la notebook 03 me di cuenta de un eror grabe
Me di cuenta del error al inspeccionar el eje vertical (Y) y el eje horizontal (X) del mapa de calor. En lugar de ver una sola fila limpia para cada país, la matriz se fragmentó en múltiples filas redundantes debido a diferencias de formato en los strings (como Argentina, argentina, ARG o nombres con guiones sobrantes). Esta dispersión sesga los datos, ya que el conteo real de un país queda dividido erróneamente en cuatro o cinco filas independientes en lugar de consolidarse en una sola